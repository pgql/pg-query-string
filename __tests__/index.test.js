import squel from '../src';

it('select', () => {
  expect(
    squel
      .select()
      .field('id')
      .from('students')
      .where(
        squel
          .expr()
          .and("name = 'Thomas'")
          .or('age > 18')
      )
      .toParam()
  ).toMatchSnapshot();
});

it('insert', () => {
  expect(
    squel
      .insert()
      .into('users')
      .set('username', 'pyramation', 'type1')
      .set('email', 'pyramation@gmail.com', 'type2')
      .returning('*')
      .toParam({ casts: { username: 'int' } })
  ).toMatchSnapshot();
});

it('update', () => {
  expect(
    squel
      .update()
      .table('students')
      .set('name', 'Fred', 'text')
      .set('age', 29)
      .set('data', { a: 1, b: 2 }, 'jsonb')
      .set('score', 1.2, 'numeric')
      .set('graduate', false, 'boolean')
      .set(
        'level',
        squel
          .select()
          .field('MAX(level)')
          .from('levels')
      )
      .toParam()
  ).toMatchSnapshot();
});

it('where', () => {
  expect(
    squel
      .select()
      .field('id')
      .from('students')
      .where(
        squel
          .expr()
          .and('name = ?', 'dude')
          .or('age > ?', 18)
      )
      .limit(2)
      .toParam()
  ).toMatchSnapshot();
});

it('delete', () => {
  expect(
    squel
      .delete()
      .from('students')
      .where('id > ?', 1)
      .where('id < ?', 100)
      .order('id', false)
      .limit(5)
      .toParam()
  ).toMatchSnapshot();
});
