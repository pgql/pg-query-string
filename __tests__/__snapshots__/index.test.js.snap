// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete 1`] = `
Object {
  "text": "DELETE FROM students WHERE (id > $1) AND (id < $2) ORDER BY id DESC LIMIT $3",
  "values": Array [
    1,
    100,
    5,
  ],
}
`;

exports[`insert 1`] = `
Object {
  "text": "INSERT INTO users (username, email) VALUES ($1::type1, $2::type2) RETURNING *",
  "values": Array [
    "pyramation",
    "pyramation@gmail.com",
  ],
}
`;

exports[`select 1`] = `
Object {
  "text": "SELECT id FROM students WHERE (name = 'Thomas' OR age > 18)",
  "values": Array [],
}
`;

exports[`update 1`] = `
Object {
  "text": "UPDATE students SET name = $1::text, age = $2, data = $3::jsonb, score = $4::numeric, graduate = $5::boolean, level = (SELECT MAX(level) FROM levels)",
  "values": Array [
    "Fred",
    29,
    Object {
      "a": 1,
      "b": 2,
    },
    1.2,
    false,
  ],
}
`;

exports[`where 1`] = `
Object {
  "text": "SELECT id FROM students WHERE (name = $1 OR age > $2) LIMIT $3",
  "values": Array [
    "dude",
    18,
    2,
  ],
}
`;
